init.lua

```diff
@@ -67,10 +106,21 @@ function craftguide:get_formspec(player_name, pagenum, recipe_num)
 			local X = (i-1) % width + 4.5
 			local Y = math.floor((i-1) / width + (6 - math.min(2, rows)))
 			local label = ""
-			if v:sub(1,6) == "group:" then label = "\nG" end
+			local tooltip = ""
+			local groupstring_inner, groupcount = craftguide.extract_groupnames(v)
+			if groupcount > 0 then
+				label = "\nG"
+				local groupstring
+				if groupcount > 1 then
+					groupstring = string.format("Any item belonging to the following groups: %s", groupstring_inner)
+				else
+					groupstring = string.format("Any item belonging to the %s group", groupstring_inner)
```

Comment:
I just merged some pre-0.4.15 functions, since the engine's release is close. You can go ahead with it.